deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java

```diff
@@ -218,7 +218,17 @@ public Response create(
         if(buildBean.getScm_branch().equals("private") && ! envBean.getAllow_private_build()) {
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, String.format("This stage does not allow deploying a private build. Please Contact #teletraan and #security-related to whitelist your stage for deploying private build"));
         }
+
+        // check if the build is from a valid artifact url
+        if(!buildBean.getArtifact_url().startsWith("https://soxrepo.pinadmin.com") && !buildBean.getArtifact_url().startsWith("https://deployrepo.pinadmin.com/")) {
```

Comment:
for non open source case, this will prevent any deploy. may need to support that case.

Reply:
Removed that case to unblock normal deploy. Thanks for pointing out.